002 Package Strucure and Code Organization

manoranjith opened the pull request:
Proposal for package structure and code organization. Relates to direct-state-transfer/dst-go#38

choeppler commented:
@manoranjith, I guess 53a099f slipped into this PR by mistake, didn't it?

manoranjith commented:
Yes, commit 53a099f slipped in by mistake. I have removed it now.

choeppler commented on design/002-Package-Structure-Code-Organiation.md:
```diff
@@ -0,0 +1,90 @@
+<!-- This is a template for proposing design changes to the dst-go project. -->
+
+# Proposal: <!-- Package structure and code organization for dst-go -->
```
The proposal title should not be commented out.
```suggestion
# Proposal: Package structure and code organization for dst-go
```

manoranjith replied:
Updated the changes in this commit - fafe44f.

manoranjith commented:
Hi @choeppler, 
I have squashed the commits and updated the pull request.
I have also changed the status to `accept` and added source file for the diagram.

Can you please review it again.
